### PR TITLE
[compiler-rt] DumpAllRegisters implementation for windows arm64.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Review by vitalybuka — APPROVED** [review]

**Comment by glandium** [comment]
This fails to build on downstream builders:
```
/builds/worker/fetches/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_win.cpp(1103,31):
error: no member named 'X29' in '_CONTEXT'
 1103 |   Printf("x29 = %llx  ", ctx->X29);
      |                          ~~~  ^
/builds/worker/fetches/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_win.cpp(1104,31):
error: no member named 'X30' in '_CONTEXT'
 1104 |   Printf("x30 = %llx  ", ctx->X30);
      |                          ~~~  ^
/builds/worker/fetches/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_win.cpp(1105,31):
error: no member named 'X31' in '_CONTEXT'
 1105 |   Printf("x31 = %llx  ", ctx->X31);
      |                          ~~~  ^
3 errors generated.
```

**Comment by glandium** [comment]
The winnt.h header in the windows 10 SDK 10.0.22621 has this:
```
    /* +0x008 */ union {
                    struct {
                        ...
                        DWORD64 X28;
    /* +0x0f0 */        DWORD64 Fp;
    /* +0x0f8 */        DWORD64 Lr;
                    } DUMMYSTRUCTNAME;
                    DWORD64 X[31];
                 } DUMMYUNIONNAME;
    /* +0x100 */ DWORD64 Sp;
    /* +0x108 */ DWORD64 Pc;
```

**Comment by devnexen** [comment]
Looking at it. thanks

File: compiler-rt/test/sanitizer_common/TestCases/Windows/dump_registers_aarch64.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  RaiseException(EXCEPTION_ACCESS_VIOLATION, 0, 0, NULL);
+  // CHECK-DUMP: Register values
+  // CHECK-DUMP-NEXT: x0 = {{0x[0-9a-f]+}}   x1 = {{0x[0-9a-f]+}}   x2 = {{0x[0-9a-f]+}}   x3 = {{0x[0-9a-f]+}}
+  // CHECK-DUMP-NEXT: x4 = {{0x[0-9a-f]+}}   x5 = {{0x[0-9a-f]+}}   x6 = {{0x[0-9a-f]+}}   x7 = {{0x[0-9a-f]+}}
+  // CHECK-DUMP-NEXT: x8 = {{0x[0-9a-f]+}}   x9 = {{0x[0-9a-f]+}}  x10 = {{0x[0-9a-f]+}}  x11 = {{0x[0-9a-f]+}}
+  // CHECK-DUMP-NEXT:x12 = {{0x[0-9a-f]+}}  x13 = {{0x[0-9a-f]+}}  x14 = {{0x[0-9a-f]+}}  x15 = {{0x[0-9a-f]+}}
+  // CHECK-DUMP-NEXT:x16 = {{0x[0-9a-f]+}}  x17 = {{0x[0-9a-f]+}}  x18 = {{0x[0-9a-f]+}}  x19 = {{0x[0-9a-f]+}}
+  // CHECK-DUMP-NEXT:x20 = {{0x[0-9a-f]+}}  x21 = {{0x[0-9a-f]+}}  x22 = {{0x[0-9a-f]+}}  x23 = {{0x[0-9a-f]+}}
+  // CHECK-DUMP-NEXT:x24 = {{0x[0-9a-f]+}}  x25 = {{0x[0-9a-f]+}}  x26 = {{0x[0-9a-f]+}}  x27 = {{0x[0-9a-f]+}}
+  // CHECK-DUMP-NEXT:x28 = {{0x[0-9a-f]+}}   fp = {{0x[0-9a-f]+}}   lr = {{0x[0-9a-f]+}}   sp = {{0x[0-9a-f]+}}
```
Comment (mstorsjo) [inline compiler-rt/test/sanitizer_common/TestCases/Windows/dump_registers_aarch64.cpp:20]: This test can't possibly succeed - as you're printing `x29` etc above, while looking for the string `fp` here?

Also even with #112305 when the code actually does compile, it still won't match this test.

Please make sure that you test compile code contributions, and for test additions, that you actually have tested running them and verified that they succeed. If you did try to run the test, was it executed, or was the `REQUIRES` line not really matched?

Comment (devnexen) [reply compiler-rt/test/sanitizer_common/TestCases/Windows/dump_registers_aarch64.cpp:20]: In fact I do not have windows for arm64 just x86_64 but have access to the CONTEXT struct definitions

Comment (devnexen) [reply compiler-rt/test/sanitizer_common/TestCases/Windows/dump_registers_aarch64.cpp:20]: going to fix the test..

**Pull request merged**